### ISSUE TITLE
DOC: Add a page to help users transition from tf.keras.wrappers.scikit_learn

### ISSUE THREAD

**Issue by adriangb** [adriangb/scikeras#124]
We should have a page noting the differences (not features) vs. the old wrappers. I think so far we only have:

- `fit`/`predict` arguments should be passed to the constructor instead of as kwargs to `fit`/`predict`.
- Renamed `build_fn` to `model`
- No longer need to declare defaults in `build_fn`, instead they should be declared in the SciKeras constructor
- To get automatic one-hot encoding with `categorical_crossentropy` `loss="categorical_crossentropy"` (or `loss=CategoricalCrossentropy`, etc.) needs to be passed to the SciKeras constructor (even for user-compiled models) since we no longer introspect compiled models.

**Comment by stsievert** [comment]
I thought SciKeras aimed to be compatible with `tf.keras.wrappers.scikit_learn`?

**Comment by adriangb** [comment]
Yes, these are minor changes. I think all of these we've discussed together previously. An example with all of these changes (copied from #112):

```python3
def build_model(optimizer='adam'):
    model = Sequential()
    model.add(Flatten(input_shape=(28, 28)))
    model.add(Dense(512, activation='relu'))
    model.add(Dropout(0.2))
    model.add(Dense(10, activation='softmax'))
    model.compile(optimizer=optimizer,
              loss='categorical_crossentropy',
              metrics=['accuracy'])

    return model
    
model = KerasClassifier(build_fn=build_model)
```

New:
```python3
def build_model(optimizer):
    model = Sequential()
    model.add(Flatten(input_shape=(28, 28)))
    model.add(Dense(512, activation='relu'))
    model.add(Dropout(0.2))
    model.add(Dense(10, activation='softmax'))
    model.compile(optimizer=optimizer,
              loss='categorical_crossentropy',
              metrics=['accuracy'])

    return model
    
model = KerasClassifier(model=build_model, optimizer="rmsprop", loss="categorical_crossentropy")
```

**Comment by stsievert** [comment]
> To get automatic one-hot encoding with categorical_crossentropy loss="categorical_crossentropy" (or loss=CategoricalCrossentropy, etc.) needs to be passed to the SciKeras constructor (even for user-compiled models) since we no longer introspect compiled models.

Why not specify a default of `loss="categorical_crossentropy"` for KerasClassifier? That'd also resolve #112, correct? Does it have any side effects?

> * fit/predict arguments should be passed to the constructor instead of as kwargs to fit/predict.
> * Renamed build_fn to model

I like the implementation around `build_fn`: it still works, but issues a warning if specified. Could the same be done for parameters passed to `fit`/`predict`?

``` python
def fit(self, X, y, classes=None, ..., **kwargs):
    for k, v in kwargs.items():
        warn(f"Set fit arguments at initialization (i.e., `BaseWrapper({k}={v})`)")
    self.set_parmas(**kwargs)
```

**Comment by adriangb** [comment]
Could the same be done for parameters passed to `fit`/`predict`?
> 
> ```python
> def fit(self, X, y, classes=None, ..., **kwargs):
>     for k, v in kwargs.items():
>         warn(f"Set fit arguments at initialization (i.e., `BaseWrapper({k}={v})`)")
>     self.set_parmas(**kwargs)
> ```

We could do that, I don't see why not. But I'd like to deprecate that relatively soon.

**Comment by adriangb** [comment]
> ```python
> def fit(self, X, y, classes=None, ..., **kwargs):
>     for k, v in kwargs.items():
>         warn(f"Set fit arguments at initialization (i.e., `BaseWrapper({k}={v})`)")
>     self.set_params(**kwargs)
> ```

The main problem with this approach is that subsequent calls to `fit/predict` will use the `**kwargs`, even if they are not passed the second call. But maybe this is enough of an edge case to not worry about.
